lib/driver.rb

```diff
@@ -0,0 +1,83 @@
+require_relative 'csv_record'
+
+module RideShare
+  class Driver < CsvRecord
+
+  attr_reader :name, :vin, :trips, :status
+  
+  def initialize(id:, name:, vin:, status: :AVAILABLE, trips: nil)
+    super(id)
+
+    @name = name
+    @vin = vin
+    @status = status.to_sym
+    @trips = trips || []
+  
+    vin_pattern = /^[A-Z0-9]{17}$/i
+    if (vin_pattern =~ @vin) == nil
+      raise ArgumentError, "Invalid VIN"
+    end
+  end
+
+  def add_trip(trip)
+    @trips << trip
+  end
+
+  def change_status
```

Comment:
The name `change_status` is misleading since it just sets the status to `:UNAVAILABLE`.

```suggestion
  def make_unavailable
```